core/src/main/java/org/elasticsearch/index/query/MatchNoneQueryParser.java

```diff
@@ -36,12 +36,28 @@
     public MatchNoneQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException {
         XContentParser parser = parseContext.parser();
 
-        XContentParser.Token token = parser.nextToken();
-        if (token != XContentParser.Token.END_OBJECT) {
-            throw new ParsingException(parser.getTokenLocation(), "[match_none] query malformed");
+        String currentFieldName = null;
+        XContentParser.Token token;
+        String queryName = null;
+        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
+        while (((token = parser.nextToken()) != XContentParser.Token.END_OBJECT && token != XContentParser.Token.END_ARRAY)) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if ("_name".equals(currentFieldName)) {
+                    queryName = parser.text();
+                } else if ("boost".equals(currentFieldName)) {
+                    boost = parser.floatValue();
```

Comment:
I think that this discussion translates to fixes that have to be made to MatchNoneQueryBuilder separately... regardless of this PR. I hadn't thought about the boost, I am fine about keeping _name as well. Also, this query was added during the query refactoring because needed internally, it ended up being exposed to the outside world as well, maybe we should also document it then... :)

@cbuescher I see in the description of the PR that you mention also match_all, but there are no changes on that query, I guess match_none was the only problem then?